lmdeploy/pytorch/backends/default/token_dispatcher.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Tuple
+
+import torch
+
+from ..token_dispatcher import TokenDispatcherImpl
+
+
+class AlltoAllTokenDispatcher(TokenDispatcherImpl):
+
+    def __init__(
+        self,
+        ep_group,
+        num_experts,
+        num_local_experts: int,
+    ) -> None:
+        self.num_local_experts = num_local_experts
+        assert num_experts is not None
+        self.num_experts = num_experts
+        assert self.num_local_experts > 0, 'Expected at least one expert'
+        self.ep_size = num_experts // num_local_experts
+        self.ep_group = ep_group
+        self.tp_size = 1
+        self.input_splits = None
+        self.output_splits = None
+        self.permute_idx_device = torch.device('cuda')
+        input_chunk_idxs = torch.arange(self.num_experts, device=self.permute_idx_device)
+        self.sort_input_by_local_experts = input_chunk_idxs.reshape(-1, self.num_local_experts).T.ravel()
```

Comment:
sort_input_by_local_experts is used in `sort_chunks_by_idxs` as `sorted_idxs` and it would be synchronized. Why not make it a cpu tensor?